library/Haste/Image/Image.php

```diff
@@ -202,4 +202,252 @@ public static function addWatermark($image, $watermark, $position = 'br', $targe
         // Return the path to new image
         return $strCacheName;
     }
+
+    /**
+     * Gets a gallery string based on the typical Contao template.
+     *
+     * Possible options keys:
+     *
+     *  - template (default is gallery_default)
+     *
+     * See findImages() for more option keys.
+     *
+     * @param array $options
+     *
+     * @return string
+     */
+    public static function getGallery(array $options)
+    {
+        $images = static::findImages($options);
+
+        if (isset($options['sortBy'])) {
+            static::sortImages($images, $options['sortBy'], $options);
+        }
+
+        $body = [];
+
+        foreach ($images as $k => $image) {
+            $stdClass = $image['templateData'];
+            $stdClass->class = $k % 2 === 0 ? 'odd' : 'even';
+            $body['image'][] = $stdClass;
+        }
+
+        $template = new \FrontendTemplate($options['template'] ?: 'gallery_default');
+        $template->body = $body;
+        return $template->parse();
```

Comment:
I don't like this approach. You should just return the array of data and let user decide what template to use, of course in other template:

```php
// mod_foobar.html5
<?php $this->insert('gallery_default', Image::getGallery()) ?>
```